graph/coloring/backtracking.go

```diff
@@ -0,0 +1,47 @@
+package coloring
```

Comment:
Is adding a new `coloring` package necessary? I don't think so but if you can explain your reasoning, it might be ok. Let me know your thoughts.

Reply:
I wanted to have all the implementations collectively in a single place, that's why I created a separate package.

Reply:
Yeah, fair point. I thought since you were using a linear undirected graph, it could still be considered part of the graph package.